continuous_integration/scripts/test_report.py

```diff
@@ -102,6 +102,44 @@ def maybe_get_next_page_path(response: requests.Response) -> str | None:
     return next_page_path
 
 
+def get_jobs(workflow):
+    page = 1
+    total_count = None
+    with shelve.open("test_report_jobs") as cache:
+        url = workflow["jobs_url"]
+        try:
+            jobs = cache[url]
+        except KeyError:
+            jobs = []
+            while total_count is None or len(jobs) < total_count:
+                resp = get_from_github(
+                    workflow["jobs_url"], {"per_page": 100, "page": page}
+                ).json()
+                if total_count is None:
+                    total_count = resp["total_count"]
+                jobs.extend(resp["jobs"])
+                page += 1
+            cache[url] = jobs
+
+    df_jobs = pandas.DataFrame.from_records(jobs)
+    extracted = df_jobs.name.str.extract(
+        r"\(([\w\-]+), (\d\.\d+),\s([\w|\s]+)\)"
+    ).dropna()
+    df_jobs["OS"] = extracted[0]
+    df_jobs["python_version"] = extracted[1]
+    df_jobs["partition"] = extracted[2]
+    # We later need to join on this. Somehow the job ID is not part of the workflow schema and we have no other way to join
```

Comment:
Yeah, I think the pain of this is why I didn't do it in the first place. I think the reason is that artifacts are organized on a per-workflow basis. More than one job could upload to the same artifact name, so you can't necessarily determine which job is responsible for a given artifact.

Reply:
let's see how this evolves. We're not changing this frequently and I think the benefit of having links directly is worth the hassle.
I can't remember how much time it took me to always look up the proper job the error was referring to..

I'm sure there is a way to directly correlate this but I didn't want to invest the time fully understanding their data schema